cores/esp32/chip-debug-report.cpp

```diff
@@ -0,0 +1,338 @@
+    case EFUSE_RD_CHIP_VER_PKG_ESP32D0WDQ6  : log_printf("D0WD-Q6"); break;
+    case EFUSE_RD_CHIP_VER_PKG_ESP32D0WDQ5  : log_printf("D0WD-Q5"); break;
+    case EFUSE_RD_CHIP_VER_PKG_ESP32D2WDQ5  : log_printf("D2WD-Q5"); break;
+    case EFUSE_RD_CHIP_VER_PKG_ESP32U4WDH   : log_printf("U4WD-H"); break;
+    case EFUSE_RD_CHIP_VER_PKG_ESP32PICOD4  : log_printf("PICO-D4"); break;
+    case EFUSE_RD_CHIP_VER_PKG_ESP32PICOV302: log_printf("PICO-V3-02"); break;
+  }
+#elif CONFIG_IDF_TARGET_ESP32S2
+  uint32_t pkg_ver = REG_GET_FIELD(EFUSE_RD_MAC_SPI_SYS_3_REG, EFUSE_PKG_VERSION);
+  switch (pkg_ver) {
+    case 1: log_printf("FH16"); break;
+    case 2: log_printf("FH32"); break;
+    default: log_printf("%lu", pkg_ver); break;
+  }
+#elif CONFIG_IDF_TARGET_ESP32S3 || CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32C6
+  uint32_t pkg_ver = REG_GET_FIELD(EFUSE_RD_MAC_SPI_SYS_3_REG, EFUSE_PKG_VERSION);
+  log_printf("%lu", pkg_ver);
+#elif CONFIG_IDF_TARGET_ESP32C2
+  uint32_t pkg_ver = REG_GET_FIELD(EFUSE_RD_BLK2_DATA1_REG, EFUSE_PKG_VERSION);
+  log_printf("%lu", pkg_ver);
+#elif CONFIG_IDF_TARGET_ESP32H2
+  uint32_t pkg_ver = REG_GET_FIELD(EFUSE_RD_MAC_SYS_4_REG, EFUSE_PKG_VERSION);
+  log_printf("%lu", pkg_ver);
+#else
+  log_printf("Unknown");
+#endif
+  log_printf("\n");
+}
+
+static void printChipInfo(void){
+  esp_chip_info_t info;
+  esp_chip_info(&info);
+  log_printf("Chip Info:\n");
+  log_printf("------------------------------------------\n");
+  log_printf("  Model             : ");
+  switch(info.model){
+    case CHIP_ESP32: log_printf("ESP32\n"); break;
+    case CHIP_ESP32S2: log_printf("ESP32-S2\n"); break;
+    case CHIP_ESP32S3: log_printf("ESP32-S3\n"); break;
+    case CHIP_ESP32C2: log_printf("ESP32-C2\n"); break;
+    case CHIP_ESP32C3: log_printf("ESP32-C3\n"); break;
+    case CHIP_ESP32C6: log_printf("ESP32-C6\n"); break;
+    case CHIP_ESP32H2: log_printf("ESP32-H2\n"); break;
+    default: log_printf("Unknown %d\n", info.model); break;
+  }
+  printPkgVersion();
+  log_printf("  Revision          : ");
+  if(info.revision > 0xFF){
+    log_printf("%d.%d\n", info.revision >> 8, info.revision & 0xFF);
+  } else {
+    log_printf("%d\n", info.revision);
+  }
+  log_printf("  Cores             : %d\n", info.cores);
+  rtc_cpu_freq_config_t conf;
+  rtc_clk_cpu_freq_get_config(&conf);
+  log_printf("  Frequency         : %lu MHz\n", conf.freq_mhz);
+  log_printf("  Embedded Flash    : %s\n", (info.features & CHIP_FEATURE_EMB_FLASH)?"Yes":"No");
+  log_printf("  Embedded PSRAM    : %s\n", (info.features & CHIP_FEATURE_EMB_PSRAM)?"Yes":"No");
+  log_printf("  2.4GHz WiFi       : %s\n", (info.features & CHIP_FEATURE_WIFI_BGN)?"Yes":"No");
+  log_printf("  Classic BT        : %s\n", (info.features & CHIP_FEATURE_BT)?"Yes":"No");
+  log_printf("  BT Low Energy     : %s\n", (info.features & CHIP_FEATURE_BLE)?"Yes":"No");
+  log_printf("  IEEE 802.15.4     : %s\n", (info.features & CHIP_FEATURE_IEEE802154)?"Yes":"No");
```

Comment:
I think including the info "Ethernet: Yes/No" will also be good.

Reply:
this info comes from the efuses, while other "features" like available peripherals are something very different. We know that only ESP32 has ethernet. This info is more for us than to tell the user what they have on-board.